crates/fluvio-smartengine/src/engine/mod.rs

```diff
@@ -0,0 +1,17 @@
+pub mod metrics;
+/// SmartModule configuration
+mod config;
+pub use config::{
+    SmartModuleConfig, SmartModuleConfigBuilder, SmartModuleConfigBuilderError,
+    SmartModuleInitialData,
+};
+mod error;
+
+#[cfg(test)]
+mod fixture;
+
+pub type WasmSlice = (i32, i32, u32);
+pub type Version = i16;
+
+mod wasmtime;
+pub use self::wasmtime::{SmartEngine, SmartModuleChainBuilder, SmartModuleChainInstance};
```

Comment:
this should be gated using feature flag wasmtime

Reply:
Hmmm, I thought you don't want to change that. If you want to add new flags for different engines, why not review #3151 directly? This PR is intended to be minimal.

Reply:
I would suggest merge this PR AS-IS, and discuss more about such things in https://github.com/infinyon/fluvio/pull/3151

Reply:
To clarify, this PR hasn't considered "how to add alternative engines", and just contains *trivial* refactoring to make that one step easier. So I don't want to add new feature flags here. And I think how to add feature flags can be nontrivial and break things and might need further discussion, so I'd rather discuss it in a PR where we introduce alternative engines.

Reply:
K. That's fair enough.  Let's do min changes here